trunk/SUAVE/Components/Energy/Storages/Batteries/Constant_Mass/Lithium_Ion_LiFePO4_18650.py

```diff
@@ -0,0 +1,246 @@
+               
+             inputs.
+                   I_bat             (current)                             [amps]
+                   P_bat             (power)                               [Watts]
+           
+           Outputs:
+             battery.          
+                  current_energy                                           [Joules]
+                  cell_temperature                                         [Kelvin]
+                  resistive_losses                                         [Watts] 
+                  load_power                                               [Watts]
+                  current                                                  [Amps]
+                  battery_voltage_open_circuit                             [Volts]
+                  battery_thevenin_voltage                                 [Volts]
+                  charge_throughput                                        [Amp-hrs]
+                  internal_resistance                                      [Ohms]
+                  battery_state_of_charge                                  [unitless]
+                  depth_of_discharge                                       [unitless]
+                  battery_voltage_under_load                               [Volts]   
+            
+        """ 
+         
+        # Unpack varibles 
+        battery           = self
+        I_bat             = battery.inputs.current
+        P_bat             = battery.inputs.power_in  
+        R_bat             = battery.resistance
+        V_max             = battery.max_voltage
+        bat_mass          = battery.mass_properties.mass                
+        bat_Cp            = battery.specific_heat_capacity    
+        T_current         = battery.pack_temperature   
+        E_max             = battery.max_energy
+        E_current         = battery.current_energy 
+        Q_prior           = battery.charge_throughput 
+        R_growth_factor   = battery.R_growth_factor
+        E_growth_factor   = battery.E_growth_factor 
+        I                 = numerics.time.integrate 
+        
+        # ---------------------------------------------------------------------------------
+        # Compute battery electrical properties 
+        # --------------------------------------------------------------------------------- 
+        n_parallel        = battery.pack_config.parallel  
+         
+        # Update battery capacitance (energy) with aging factor
+        E_max = E_max*E_growth_factor
+        
+        # Compute state of charge and depth of discarge of the battery
+        initial_discharge_state = np.dot(I,P_bat) + E_current[0]
+        DOD_old                 = 1 - np.divide(initial_discharge_state,E_max)
+        DOD_old[DOD_old< 0.]    = 0. 
+        
+        # compute the C rate
+        C = np.abs(3600.*P_bat/E_max)
+        
+        # Empirical for for discharge   
+        f = 1-np.exp(-20.*DOD_old)-np.exp(-20.*(1.- DOD_old)) 
```

Comment:
FYI this equation was the one we discussed today. There are other fits

Reply:
Added new resistance function along with reference